### DISCUSSION TITLE
Getting error while reading netcdf file

### DISCUSSION THREAD

**Discussion by ghost** [dtcenter/METplus#2103] [File I/O] [answered]
Hi,
I am trying to read model output file and getting error while trying plot_data_plane. Netcdf files are on Hera at : 
/scratch1/NCEPDEV/stmp2/Partha.Bhattacharjee/RRFSSD_Datatest
And I tried  plot_data_plane phyf000.nc test.ps 'name="aodtot"; level="(0,*,*)";' -v 7
 Opening data file: phyf000.nc
DEBUG 4: Met2dDataFileFactory::new_met_2d_data_file() -> created new Met2dDataFile object of type "FileType_None".
ERROR  : 
ERROR  : plot_data_plane -> file "phyf000.nc" not a valid data file

Any help on how to read that using Met. Thanks in advance.

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hi @perthsb, I hopped on hera and took a look at the NetCDF file you're trying to read into MET:
/scratch1/NCEPDEV/stmp2/Partha.Bhattacharjee/RRFSSD_Datatest/phyf000.nc

I see the same behavior that you see when processing this through plot_data_plane. Looking at the header (below), I see that this file is not formatted following any of the NetCDF conventions that MET currently supports (CF-compliant, MET, or WRF formatting).
```
> ncdump -h phyf000.nc
// global attributes:
		:cen_lat = 38.5f ;
		:cen_lon = -97.5f ;
		:dtp = 60.f ;
		:dx = 3000.f ;
		:dy = 3000.f ;
		:fhzero = 1 ;
		:grid = "lambert_conformal" ;
		:grid_id = 1 ;
		:imp_physics = 8 ;
		:landsfcmdl = 3 ;
		:lat1 = 21.13812f ;
		:lon1 = -122.7195f ;
		:ncld = 8 ;
		:nsoil = 4 ;
		:nx = 1799 ;
		:ny = 1059 ;
		:source = "FV3GFS" ;
		:stdlat1 = 38.5f ;
		:stdlat2 = 38.5f ;
```
So this isn't in a format supported directly by MET.

How was this file created? I see `source = "FV3GFS" ;` which indicates that its from FV3 but I didn't think FV3 wrote `lambert_conformal` output. Does this file follow some documented NetCDF convention or is it custom?

We could ingest this into MET using python embedding but not directly at this time.

**Reply by ghost** [reply]
Thanks @JohnHalleyGotway for looking into this. This is output from RRFS (Rapid Refresh Forecast System) and it is over CONUS written in Lambert_conformal grid at 3 km resolution (similar to HRRR files).
 Interestingly, RRFS also produces grib2 output using UPP from the same model run that creates the netcdf file I sent you earlier and those grib2 files MET can read (same Lambert_conformal). One example is : 

plot_data_plane /scratch1/NCEPDEV/stmp4/Partha.Bhattacharjee/RRFSSD_Jet/DustRetro_20230226/AerofeedONCor/Grib2file/RRFS_CONUS.t00z.bgdawpf000.tm00.grib2 test.ps 'name="AOTK"; level="R806";' -v 7

 Opening data file: RRFS_CONUS.t00z.bgdawpf000.tm00.grib2
DEBUG 4: Met2dDataFileFactory::new_met_2d_data_file() -> created new Met2dDataFile object of type "FileType_Gb2".
DEBUG 4: Switching the GRIB2 radius of the earth value of 6371.23 km to 6371.2 km for internal consistency.
DEBUG 4: 
DEBUG 4: Lambert Conformal Grid Data:
DEBUG 4:    hemisphere: N
DEBUG 4:   scale_lat_1: 38.5
DEBUG 4:   scale_lat_2: 38.5
DEBUG 4:       lat_pin: 21.1381
DEBUG 4:       lon_pin: 122.72
DEBUG 4:         x_pin: 0
DEBUG 4:         y_pin: 0
DEBUG 4:    lon_orient: 97.5
DEBUG 4:          d_km: 3
DEBUG 4:          r_km: 6371.2
DEBUG 4:            nx: 1799
DEBUG 4:            ny: 1059
DEBUG 4:     so2_angle: 0
DEBUG 4: 
DEBUG 4: MetGrib2DataFile::read_grib2_record_list() - unrecognized GRIB2 field indexes - disc: 0, master table: 2, center: 7, local table: 1, parm_cat: 7, parm: 200
DEBUG 4: MetGrib2DataFile::read_grib2_record_list() - unrecognized GRIB2 field indexes - disc: 0, master table: 2, center: 7, local table: 1, parm_cat: 7, parm: 200
DEBUG 4: MetGrib2DataFile::read_grib2_record_list() - unrecognized GRIB2 field indexes - disc: 1, master table: 2, center: 7, local table: 1, parm_cat: 0, parm: 9
DEBUG 4: MetGrib2DataFile::read_grib2_record_list() - unrecognized GRIB2 field indexes - disc: 1, master table: 2, center: 7, local table: 1, parm_cat: 0, parm: 9
DEBUG 4: MetGrib2DataFile::read_grib2_record_list() - unrecognized GRIB2 field indexes - disc: 1, master table: 2, center: 7, local table: 1, parm_cat: 0, parm: 9
DEBUG 4: MetGrib2DataFile::read_grib2_record_list() - unrecognized GRIB2 field indexes - disc: 1, master table: 2, center: 7, local table: 1, parm_cat: 0, parm: 9
DEBUG 4: MetGrib2DataFile::read_grib2_record_list() - unrecognized GRIB2 field indexes - disc: 1, master table: 2, center: 7, local table: 1, parm_cat: 0, parm: 9
DEBUG 4: MetGrib2DataFile::read_grib2_record_list() - unrecognized GRIB2 field indexes - disc: 1, master table: 2, center: 7, local table: 1, parm_cat: 0, parm: 9
DEBUG 4: MetGrib2DataFile::read_grib2_record_list() - unrecognized GRIB2 field indexes - disc: 1, master table: 2, center: 7, local table: 1, parm_cat: 0, parm: 9
DEBUG 4: MetGrib2DataFile::read_grib2_record_list() - unrecognized GRIB2 field indexes - disc: 1, master table: 2, center: 7, local table: 1, parm_cat: 0, parm: 9
DEBUG 4: MetGrib2DataFile::read_grib2_record_list() - unrecognized GRIB2 field indexes - disc: 2, master table: 2, center: 7, local table: 1, parm_cat: 4, parm: 5
DEBUG 4: MetGrib2DataFile::read_grib2_record_list() - unrecognized GRIB2 field indexes - disc: 0, master table: 2, center: 7, local table: 1, parm_cat: 0, parm: 205
DEBUG 4: MetGrib2DataFile::read_grib2_record_list() - processed 988 records from GRIB2 file: RRFS_CONUS.t00z.bgdawpf000.tm00.grib2
DEBUG 4: VarInfoFactory::new_var_info() -> created new VarInfo object of type "FileType_Gb2".
DEBUG 3: MetGrib2DataFile::data_plane() -> Found exact match for 'AOTK/R806' in GRIB2 record 806 field 1 of GRIB2 file 'RRFS_CONUS.t00z.bgdawpf000.tm00.grib2'
DEBUG 4: 
DEBUG 4: Data plane information:
DEBUG 4:       plane min: 0
DEBUG 4:       plane max: 0.0034212
DEBUG 4:       scan mode: 64
DEBUG 4:      valid time: 20230226_000000
DEBUG 4:       lead time: 000000
DEBUG 4:       init time: 20230226_000000
DEBUG 4:     bitmap flag: 255
DEBUG 4: 
DEBUG 4: 
DEBUG 4: Grid information:
DEBUG 4:    Projection: Lambert Conformal
DEBUG 4:    Nx: 1799
DEBUG 4:    Ny: 1059
DEBUG 4:    Lat_LL: 21.138
DEBUG 4:    Lon_LL: 122.720
DEBUG 4:    Lon_orient: 97.500
DEBUG 4:    Alpha: 4203.494
DEBUG 4:    Cone: 0.623
DEBUG 4:    Bx: 899.1693
DEBUG 4:    By: 3198.9984
DEBUG 4: 
DEBUG 4: Data plane information:
DEBUG 4:    plane min: 0
DEBUG 4:    plane max: 0.0034212
DEBUG 4:    valid time: 20230226_000000
DEBUG 4:    lead time: 000000
DEBUG 4:    init time: 20230226_000000
DEBUG 4:    accum time: 000000
DEBUG 4: 
DEBUG 1: Creating postscript file: test.ps
DEBUG 4: draw_map() -> Plotting map data file: /contrib/met/11.0.1/share/met/map/country_data
DEBUG 4: draw_map() -> Plotting map data file: /contrib/met/11.0.1/share/met/map/usa_state_data
DEBUG 4: draw_map() -> Plotting map data file: /contrib/met/11.0.1/share/met/map/major_lakes_data
DEBUG 1: Finish plot_data_plane by Partha.Bhattacharjee(10704) at 2023-03-30 18:19:27Z

Thanks

**Reply by JohnHalleyGotway** [reply]
OK, so I assume you're all set using the GRIB2 RRFS output instead of the NetCDF files? Is that correct?

**Reply by ghost** [reply]
Well, in this case UPP did not work properly for the fields I am looking into and that is why I started looking into nedcdf files (where the fields are ok). But that problem can be fixed with UPP update and rerun. 
Thanks
